api/backup.go

```diff
@@ -18,7 +19,64 @@ func (s *Server) BackupTargetList(w http.ResponseWriter, req *http.Request) erro
 	if err != nil {
 		return errors.Wrap(err, "failed to list backup targets")
 	}
-	apiContext.Write(toBackupTargetCollection(backupTargets))
+	apiContext.Write(toBackupTargetCollection(backupTargets, apiContext))
+	return nil
+}
+
+func (s *Server) BackupTargetUpdateAll(w http.ResponseWriter, req *http.Request) error {
+	var input BackupTarget
+
+	apiContext := api.GetApiContext(req)
+	if err := apiContext.Read(&input); err != nil {
+		return err
+	}
+
+	bts, err := s.m.ListBackupTargetsSorted()
+	if err != nil {
+		return errors.Wrap(err, "failed to list backup targets")
+	}
+
+	if input.SyncBackupTarget {
+		for _, bt := range bts {
+			if _, err := s.m.SyncBackupTarget(bt); err != nil {
+				logrus.Warnf("failed to synchronize backup target %v", bt.Name)
+			}
+		}
+	}
+
+	apiContext.Write(toBackupTargetCollection(bts, apiContext))
+	return nil
+}
+
+func (s *Server) SyncBackupTarget(w http.ResponseWriter, req *http.Request) error {
+	var input SyncBackupTarget
+
+	apiContext := api.GetApiContext(req)
+	if err := apiContext.Read(&input); err != nil {
+		return err
+	}
+
+	backupTargetName := mux.Vars(req)["backupTargetName"]
+	if backupTargetName == "" {
+		return fmt.Errorf("backup target name is required")
+	}
+
+	bt, err := s.m.GetBackupTarget(backupTargetName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get backup target %v", backupTargetName)
+	}
+
+	if input.SyncBackupTarget {
+		updatedBackupTarget, err := s.m.SyncBackupTarget(bt)
+		if err != nil {
+			logrus.Warnf("failed to synchronize backup target %v", backupTargetName)
```

Comment:
For single backuptarget, shall we error out?

Reply:
Yes, let's return the error.